### PR TITLE
Reduce the coupling of the mailservice with the rest of the application

### PR DESCRIPTION
**Pull request by jespino**
#### Summary
Reduce the coupling of the mailservice with the rest of the application

#### Ticket Link
[MM-32885](https://mattermost.atlassian.net/browse/MM-32885)

#### Related PR:
enterprise: mattermost/enterprise#924

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Comment by streamer45** [comment]
@jespino 

just so you know @isacikgoz is OoO in case you wanted this merged soon.

**Comment by jespino** [comment]
@streamer45 there is no rush, but I'll change it anyway.

File: services/mailservice/mail.go [hunk 1]
```diff
@@ -345,20 +357,6 @@ func SendMail(c smtpClient, mail mailData, fileBackend filesstore.FileBackend, d
 		m.EmbedReader(name, reader)
 	}
 
-	for _, fileInfo := range mail.attachments {
```
Comment (streamer45) [inline services/mailservice/mail.go:348]: So are we dropping support for attachments?

Comment (jespino) [reply services/mailservice/mail.go:348]: We are dropping "not exposed/not used" file attachments in the emails (except for the embedded files) that are supported and handled through other way. If we want to add back attachments in the future, I would recommend to add them without depending explicitly from the filestoreage package by itself (the coupling there wasn't needed event if we wanted to attach files).

Comment (streamer45) [reply services/mailservice/mail.go:348]: Sounds good :+1: We can always add a method in the future that takes files/readers.

**Review by streamer45 — APPROVED** [review]
Looks good :+1:

**Review by wiggin77 — APPROVED** [review]
LGTM :+1:

File: services/mailservice/mail.go [hunk 2]
```diff
@@ -18,11 +18,29 @@ import (
 	gomail "gopkg.in/mail.v2"
 
 	"github.com/mattermost/mattermost-server/v5/mlog"
-	"github.com/mattermost/mattermost-server/v5/model"
-	"github.com/mattermost/mattermost-server/v5/services/filesstore"
-	"github.com/mattermost/mattermost-server/v5/utils"
 )
 
+const (
+	connSecurityTls      = "TLS"
```
Comment (agnivade) [inline services/mailservice/mail.go:24]: TLS and StartTLS

File: services/mailservice/mail.go [hunk 3]
```diff
@@ -140,15 +157,15 @@ func ConnectToSMTPServerAdvanced(connectionInfo *SmtpConnectionInfo) (net.Conn,
 	return conn, nil
 }
 
-func ConnectToSMTPServer(config *model.Config) (net.Conn, error) {
+func ConnectToSMTPServer(config *SMTPConfig) (net.Conn, error) {
 	return ConnectToSMTPServerAdvanced(
 		&SmtpConnectionInfo{
```
Comment (agnivade) [inline services/mailservice/mail.go:162]: I am wondering if there is a need to duplicate the config in two structs? It seems like `SmtpConnectionInfo` already contains what we need. If we are exposing `SMTPConfig` to the outside world, can we just clean up `SmtpConnectionInfo` and use only one config struct throughout?

**Review by agnivade — COMMENTED** [review]
Looks great. Spotted an opportunity for further cleanup.

File: services/mailservice/mail.go (hunk 3 above)
Comment (jespino) [reply services/mailservice/mail.go:162]: Yes, that makes a lot of sense, for sure we can make it simpler.

Comment (jespino) [reply services/mailservice/mail.go]: done

File: services/mailservice/mail.go (hunk 2 above)
Comment (jespino) [reply services/mailservice/mail.go:24]: Done

File: services/mailservice/mail.go [hunk 4]
```diff
@@ -62,15 +80,15 @@ type SmtpConnectionInfo struct {
```
Comment (agnivade) [inline services/mailservice/mail.go:80]: We can actually remove the struct declaration above. L69.

**Review by agnivade — APPROVED** [review]
Looks amazing.

File: services/mailservice/mail.go (hunk 4 above)
Comment (jespino) [reply services/mailservice/mail.go:80]: :facepalm:

**Review by streamer45 — APPROVED** [review]

**Review by lindy65 — APPROVED** [review]
Thanks @jespino - Global Relay email is working as expected as are email invitations, notification emails and emails received when changing password, etc. Looks good to me 👍

**Pull request merged**